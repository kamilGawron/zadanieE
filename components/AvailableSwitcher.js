import React from "react"
const shortid = require('shortid');

export default function AvailableSwitcher(props){
    return (
        <div key={shortid.generate()}>
            <span>Tylko dostÄ™pne</span>
            <label className="switch">
                <input 
                    name="available" 
                    id="availableSwitcher" 
                    type="checkbox" 
                    onChange={props.availableToggler} 
                    defaultChecked
                />
                <span className="slider round"></span>
            </label>
        </div>
    )
}