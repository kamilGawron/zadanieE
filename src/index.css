:root{
    font-size: 10px;
}
body{
    padding: 0;
    margin: 0;
}
.loading{
    height: 100vh;
    width: 100vw;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    background-color: rgba(0,0,0,.9);
    font-size: 2rem;
    color: #ffffff;
    text-transform: uppercase;
}
.loading .spiner{
    margin-top: 30px;
    height: 80px;
    width: 80px;
    border-radius: 50%;
    border: 5px solid rgba(0,0,0,0);
    border-bottom: 5px solid #2196F3;
    -webkit-animation: spiner .7s ease-in-out;
    animation: spiner .8s ease-in-out infinite;
}
@-webkit-keyframes spiner{
    0%{
        -webkit-transform: rotate(0);
        transform: rotate(0);
    }
    100%{
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}
@keyframes spiner{
    0%{
        -webkit-transform: rotate(0);
        transform: rotate(0);
    }
    100%{
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}
.settings{
    z-index: 1000;
    position: absolute;
    top: 0;
    left: 0;
    height: 100px;
    width: 100%;
    background-color: rgba(0,0,0,.9);
    border-bottom: 2px solid #2196F3;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .general{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex; 
    -webkit-box-orient: vertical; 
    -webkit-box-direction: normal; 
    -ms-flex-direction: column; 
    flex-direction: column;
    -webkit-box-pack: center; 
    -ms-flex-pack: center; 
    justify-content: center;
    color: #ffffff;
    font-size: 1.4rem;
}
.settings .general>h6{
    margin: 5px;
    text-align: center;
    font-size: 1.6rem;
}
.settings .general>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
}
.settings .general>div>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin: 10px;
    width: 100px;
}
.settings .general div:last-of-type{
    margin-left: 50px;
    
}
.settings .general div .filters-icon{
    font-size: 2.4rem;
    cursor: pointer;
    color: #ffffff;
    -webkit-transition: all .7s ease-in-out;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .general div .filters-icon:hover{
    color: #2196F3;
}
.settings #filters{
    position: absolute;
    top:100px;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    width: 60%;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    background-color: rgba(0,0,0,.9);
    padding: 10px;
    padding-left: 50px;
    border-radius: 20px;
    color: #ffffff;
    font-size: 1.4rem;

    
}
.settings .specific-settings .close{
    position: absolute;
    right: 20px;
    top:20px;
    font-size: 2.5rem;
    font-weight: bold;
    cursor: pointer;
    color: #ffffff;
    -webkit-transition: all .7s ease-in-out;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .specific-settings .close:hover{
    color: #2196F3;
}
.settings .specific-settings input::-webkit-outer-spin-button,
.settings .specific-settings input::-webkit-inner-spin-button {
    /* display: none; <- Crashes Chrome on hover */
    -webkit-appearance: none;
    margin: 0; 
}

.settings .specific-settings input[type=number]{
    width: 40px;
    border: none;
    background-color: transparent;
    color: #ffffff;
    border-bottom: 1px solid #2196F3;
    -moz-appearance:textfield;
    text-align: center;
    outline: none;
}
.settings .specific-settings input[type=range]::-webkit-slider-thumb,
.settings .specific-settings input[type=range]::-moz-range-thumb,
.settings .specific-settings input[type=range]::-ms-thumb{
   color: red;
}

.settings .specific-settings .cars,
.settings .specific-settings .parkings{
    margin: 0;
    width: 50%;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .specific-settings .cars>div,
.settings .specific-settings .parkings>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .specific-settings .cars>div .switch{
    top: 0;
    left: 10px;
}
.settings .specific-settings .cars h6,
.settings .specific-settings .parkings h6{
    margin: 0 0 10px 0;
    font-size: 1.6rem;
}
.settings .specific-settings div>span{
    margin: 10px 0;
    font-weight: bold;
    color: #2196F3;
    font-size: 1.4rem;
}
.settings .specific-settings .clear-input{
    color: #ffffff;
    cursor: pointer;
    height: 20px;
    width: 20px;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .specific-settings .clear-input:hover{
    color: #2196F3;
}
.switch {
    position: relative;
    display: inline-block;
    width: 46px;
    height: 25px;
    top:5px;
}

.switch input {
    opacity: 0;
    width: 0;
    height: 0;
}

/* The slider */
.slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    -webkit-transition: .4s;
    transition: .4s;
}

.slider:before {
    position: absolute;
    content: "";
    height: 16px;
    width: 16px;
    left: 4px;
    bottom: 4px;
    background-color: white;
    -webkit-transition: .4s;
    transition: .4s;
}

input:checked + .slider {
    background-color: #2196F3;
}

input:focus + .slider {
    -webkit-box-shadow: 0 0 1px #2196F3;
    box-shadow: 0 0 1px #2196F3;
}

input:checked + .slider:before {
    -webkit-transform: translateX(22px);
    transform: translateX(22px);
}

/* Rounded sliders */
.slider.round {
    border-radius: 34px;
}

.slider.round:before {
    border-radius: 50%;
}

.leaflet-container{
    height: 100vh;
    width: 100vw;
    margin: 0;
}
.leaflet-control{
    top:100px;
}
.leaflet-popup-content-wrapper{
    background-color: rgba(0,0,0,.9);
    color: #ffffff;
}
.leaflet-popup-content h3{
    margin: 5px 0;
    text-align: center;
    color: #2196F3;
}
.leaflet-popup-content .car>div:first-of-type{
    text-align: center;
    font-weight: bold;
    margin:0 0 5px 0;
}
.leaflet-popup-content .content .title{
    font-weight: bold;
    margin-right: 5px;
}
.leaflet-popup-content .poi>div:first-of-type{
    text-align: center;
    margin-bottom: 5px;
    
}
.leaflet-popup-tip{
    background-color: rgba(0,0,0,.9);
}
.leaflet-container a.leaflet-popup-close-button{
    color: #ffffff;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.leaflet-container a.leaflet-popup-close-button:hover{
    color:  #2196F3;
}
@media all and (max-width:768px){
    .settings #filters{
        width: 100%;
        border-radius: 0;
    }
    .settings .specific-settings input[type=range]{
        width: 120px;
    }
    .settings .general>div{
        padding: 0;
        -webkit-box-pack: start;
        -ms-flex-pack: start;
    }
    .settings .general>div>div{
        width: 90px;
        font-size: 1.1rem;
        margin: 0px;
    }
    .settings .general div:last-of-type{
        margin-left: 0px;
        
    }
    .settings #filters{
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
    }
    .settings .specific-settings .cars,
    .settings .specific-settings .parkings{
        width: 100%;
    }
    .settings .specific-settings .parkings{
        margin-top: 10px;
    }
    
}