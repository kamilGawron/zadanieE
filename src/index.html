<!DOCTYPE html>
<html lang="">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title></title>
    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.0.1/dist/leaflet.css" />

</head>

<body>
   <div id="root"></div>
    <script src=""></script>
    
</body>
</html>
