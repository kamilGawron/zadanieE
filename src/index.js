import React from 'react'
import ReactDOM from 'react-dom'
import App from '../components/App.js'
require('../style/index.css');
require('../style/media.css');
require('../style/spinner.css');
require('../style/map.css');
require('../style/settings.css');


ReactDOM.render(<App />,document.getElementById('root'))