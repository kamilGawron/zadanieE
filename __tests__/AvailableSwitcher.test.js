import React from 'react';
import { shallow } from 'enzyme';
import AvailableSwitcher from '../components/AvailableSwitcher';
import '../src/setupTest'
import renderer from 'react-test-renderer'
const props={
    availableToggler : function(){},
}
describe('test <AvailableSwitcher />', () => {
    const wrapper = shallow(<AvailableSwitcher {...props}/>);
    it('matches the snapshot', () => {
        const tree=renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('display "only available" text', () => {
        expect(wrapper.text()).toEqual('Tylko dostÄ™pne');
    });
    it('contain a label className="switch"`', () => {
        expect(wrapper.find('label.switch')).toHaveLength(1);
    });
    it('contain an input inside label className="switch"`', () => {
        expect(wrapper.find('label.switch input')).toHaveLength(1);
    });
    it('contain a span className="slider round" inside label className="switch"`', () => {
        expect(wrapper.find('label.switch span.slider.round')).toHaveLength(1);
    });
    it('switcher is default checked', () => {
        expect(wrapper.find('label.switch input').props().defaultChecked).toEqual(true);
    });
});