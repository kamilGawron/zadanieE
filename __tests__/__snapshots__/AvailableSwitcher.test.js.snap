// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <AvailableSwitcher /> matches the snapshot 1`] = `
<div>
  <span>
    Tylko dostÄ™pne
  </span>
  <label
    className="switch"
  >
    <input
      defaultChecked={true}
      id="availableSwitcher"
      name="available"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
</div>
`;
